easybuild/tools/containers/singularity.py

```diff
@@ -256,19 +264,73 @@ def resolve_template_data(self):
                 bootstrap_config_lines.append('%s: %s' % (key, template_data[key.lower()]))
         template_data['bootstrap_config'] = '\n'.join(bootstrap_config_lines)
 
-        # if there is osdependencies in easyconfig then add them to Singularity recipe
-        install_os_deps = ''
+        # basic tools & utilities to install in container image
+        osdeps = []
+
+        # install bunch of required/useful OS packages, but only when starting from scratch;
+        # when starting from an existing image, the required OS packages are assumed to be installed already
```

Comment:
Since we are making this assumption, we should check that the input image as been created with EB. In this way we extend the validity of such assumption (and avoid potential headaches).

Reply:
How would we check that? By inspecting the image via `singularity exec`?

Note that we don't have the restriction now that the base image you want to use is available when creating the recipe, so we can't inspect it.

It's really only needed when `--container-build-image` is used, or when the generated recipe is passed to `singularity build`.

Reply:
Good enough for me